lib/phoenix_pubsub/local.ex

```diff
@@ -186,30 +189,36 @@ defmodule Phoenix.PubSub.Local do
                         read_concurrency: true, write_concurrency: true])
 
     Process.flag(:trap_exit, true)
-    {:ok, %{topics: local, pids: gc, gc_server: gc}}
+    {:ok, %{monitors: %{}, gc: gc}}
   end
 
-  def handle_call({:subscribe, pid, _topic, opts}, _from, state) do
+  def handle_call({:subscribe, pid, opts}, _from, state) do
     if opts[:link], do: Process.link(pid)
-    Process.monitor(pid)
-    {:reply, {:ok, {state.topics, state.pids}}, state}
+    {:reply, :ok, put_new_monitor(state, pid)}
+  end
+
+  def handle_call({:unsubscribe, pid, topic}, _from, %{gc: gc} = state) do
+    :ets.match_delete(gc, {pid, topic})
```

Comment:
We could get around the race by moving back to the `{ref, count}` state and using `:ets.select_delete` as the atomic op on unsubscribe. We could still offload the ets insert to the client in that case so the only cost is tracking the ref count. Thoughts?